### PR TITLE
Allow overwrite the builtin Postgres DB password

### PR DESCRIPTION
**Pull request by megian**
The default behavior of the codecentric Helm Chart does set a
predefined password, which is insecure.

## Checklist

- [x] Keep pull requests small so they can be easily reviewed.
- [x] Update the documentation.
- [ ] Categorize the PR by setting a good title and adding one of the labels:
      `bug`, `enhancement`, `documentation`, `change`, `breaking`, `dependency`
      as they show up in the changelog
- [ ] Link this PR to related issues.

### CODE REVIEW

File: class/defaults.yml [hunk 1]
```diff
@@ -116,6 +118,7 @@ parameters:
         rules: ${keycloak:monitoring:rules}
       postgresql:
         enabled: ${keycloak:database:builtin}
+        existingSecret: ${keycloak:database:existingSecret}
```
Comment (ccremer) [inline class/defaults.yml:121]: I just noticed that it's easy to overwrite the password:
```suggestion
        postgresqlPassword: '?{vaultkv:${customer:name}/${cluster:name}/keycloak/db-password}'
```
It's possible to override this even via the hierarchy, but one has to know this...

**Review by ccremer — COMMENTED** [review]
I found a way to overwrite this without needing the "existingSecret" parameter. Maybe you could consider this? or is there a disadvantage?

File: class/defaults.yml (hunk 1 above)
Comment (ccremer) [reply class/defaults.yml:121]: ah now I understand...
the string `'?{vaultkv:${customer:name}/${cluster:name}/keycloak/db-password}'` gets base64 encoded and put it like this:
```yaml
data:
  postgresql-password: P3t2YXVsdGt2OnQtc2lsZW50LXRlc3QtMTIzNC9jLWdyZWVuLXRlc3QtMTIzNC9rZXljbG9hay9kYi1wYXNzd29yZH0=
```
which is `?{vaultkv:t-silent-test-1234/c-green-test-1234/keycloak/db-password}`. But ArgoCD won't be able to read that...

Comment (megian) [reply class/defaults.yml:121]: So we need to keep it as is, if you don't have a better solution/workaround.

**Comment by megian** [comment]
Will be resolved via #13.

**Pull request closed without merging**